protobufs/livekit_models.proto

```diff
@@ -131,6 +131,7 @@ message ParticipantInfo {
   bool is_publisher = 13;
   Kind kind = 14;
   map<string, string> attributes = 15;
+  DisconnectReason disconnect_reason = 16;
```

Comment:
any chance of potentially including insight in to whether the participant is attempting a reconnect ? 
I think users have been asking about this in the past.

Reply:
That is a transient state. Can think about it. This reason would really be valid when `state = DISCONNECTED`. Reconnect indication would be orthogonal I think. What's your thinking on how that should work?

Reply:
Thinking more about the reconnecting indication.

`reconnect = 1` if successful should be (mostly) transparent. But, that is the case where server clearly knows there is an attempt to resume.

A full reconnect looks like a new connection to the server. If the participant exists in the room and a full reconnect happens, server could indicate that the participant is attempting another connection (but that would not work in cloud if connecting to different node and that node does not have the old participant's remote participant, would require sending a message to check, but that would impact every join).

Would be good to see what users want and figure out how to do that.